tests/security/fread_truncated_wav.cpp

```
@@ -0,0 +1,209 @@
+        }
+        samples[i] = sample / static_cast<float>(numChannels);
+    }
+    return true;
+}
+
+// Helper: write a minimal WAV file with specific data size
+static void writeWav16(const char* path, uint32_t sampleRate, uint16_t channels,
+                       const int16_t* data, uint32_t numSamples) {
+    FILE* f = fopen(path, "wb");
+    // RIFF header
+    fwrite("RIFF", 1, 4, f);
```

Cppcheck (2.20.0)

[warning] If resource allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfResources)